src/main/java/org/myrobotlab/service/interfaces/UtteranceListener.java

```diff
@@ -11,5 +11,28 @@ public interface UtteranceListener {
   public String getName();
 
   public void onUtterance(Utterance utterance) throws Exception;
+  
+  
+  default public void attachUtterancePublisher(UtterancePublisher publisher) {
```

Comment:
Doesn't this break the directionality of the attach?  i thought publishers always attached listeners in the new convention.. so adding this here is removing that clean definition..  (I understand the convenience, but it also makes the directional attach convention. 

when using these methods programmatically, it doesn't matter, however, it will get confusing to folks if we don't enforce that directionality of attach in the configs.